elide-core/src/main/java/com/yahoo/elide/parsers/expression/PermissionToFilterExpressionVisitor.java

```diff
@@ -91,19 +122,19 @@ public FilterExpression visitAND(ExpressionParser.ANDContext ctx) {
 
         // (NO_EVA and UserCheck) should return NO_EVA let the upper level know.
         // For example, FilterExpressionCheck Or (NO_EVA and UserCheck) should evaluate to  NO_EVALUATION_EXPRESSION.
-        if (left == NO_EVALUATION_EXPRESSION && right == USER_CHECK_EXPRESSION) {
+        if (left == NO_EVALUATION_EXPRESSION && right instanceof UserCheckExpression) {
```

Comment:
This would change as well if we encounter FALSE_USER_CHECK_EXPRESSION.  

Anything AND FALSE_USER_CHECK_EXPRESSION should return FALSE_USER_CHECK_EXPRESSION.